From: jaimergp
Date: Tue Feb 27 12:01:36 CEST 2024
Subject: Return 'unknown' instead of raising an exception when the calling module cannot be determined

diff --git a/conda/deprecations.py b/conda/deprecations.py
index 5d4c0df28..924428fa0 100644
--- a/conda/deprecations.py
+++ b/conda/deprecations.py
@@ -288,11 +288,12 @@ class DeprecationHandler:
             module = inspect.getmodule(frame)
             if module is not None:
                 return (module, module.__name__)
-        except IndexError:
+        except (IndexError, AttributeError):
             # IndexError: 2 + stack is out of range
             pass
 
-        raise DeprecatedError("unable to determine the calling module")
+        # raise DeprecatedError("unable to determine the calling module")
+        return (sys, "<unknown>")
 
     def _generate_message(
         self,
