From 42ed3ab04aca688cb4081aea5db590e40d0852d6 Mon Sep 17 00:00:00 2001
From: Jaida Rice <jaidarice@gmail.com>
Date: Thu, 4 Sep 2025 16:40:26 -0700
Subject: [PATCH] Use default=NULL for --override-frozen

---
 conda/cli/helpers.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/conda/cli/helpers.py b/conda/cli/helpers.py
index cab249d54..4c4512147 100644
--- a/conda/cli/helpers.py
+++ b/conda/cli/helpers.py
@@ -239,9 +239,12 @@ def add_output_and_prompt_options(p: ArgumentParser) -> _ArgumentGroup:


 def add_parser_frozen_env(p: ArgumentParser):
+    from ..common.constants import NULL
+
     p.add_argument(
         "--override-frozen",
         action="store_false",
+        default=NULL,
         help="DANGEROUS. Use at your own risk. Ignore protections if the environment is frozen.",
         dest="protect_frozen_envs",
     )
--
